app/src/main/java/com/bogdan/codeforceswatcher/features/auth/SignUpComposeActivity.kt

```diff
@@ -0,0 +1,298 @@
+
+        Scaffold(
+            modifier = Modifier.fillMaxSize(),
+            topBar = {
+                NavigationBar { finish() }
+            },
+            bottomBar = {
+                LinkText(
+                    linkTextData = listOf(
+                        LinkTextData("${getString(R.string.already_have_an_account)} "),
+                        LinkTextData(getString(R.string.sign_in), "sign_in") {
+                            startSignInActivity()
+                            finish()
+                        }
+                    ),
+                    modifier = Modifier
+                        .height(62.dp)
+                        .fillMaxWidth()
+                        .padding(horizontal = 20.dp),
+                    textStyle = MaterialTheme.typography.body1.copy(
+                        fontSize = 14.sp,
+                        color = MaterialTheme.colors.secondaryVariant
+                    ),
+                    clickableTextStyle = MaterialTheme.typography.body2.copy(
+                        fontSize = 14.sp,
+                        color = MaterialTheme.colors.onBackground
+                    ),
+                    paragraphStyle = ParagraphStyle(textAlign = TextAlign.Center)
+                )
+            },
+            backgroundColor = MaterialTheme.colors.background
+        ) {
+            Column(
+                modifier = Modifier.padding(horizontal = 20.dp),
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                Spacer(Modifier.height(56.dp))
+
+                Title(getString(R.string.sign_up))
+
+                Spacer(Modifier.height(44.dp))
+
+                AuthTextField(
+                    label = getString(R.string.email),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Email,
+                        imeAction = ImeAction.Next
+                    ),
+                    keyboardActions = KeyboardActions(
+                        onNext = { localFocusManager.moveFocus(FocusDirection.Down) }
+                    )
+                ) { newEmail ->
+                    email = newEmail
+                }
+
+                Spacer(Modifier.height(24.dp))
+
+                AuthTextField(
+                    label = getString(R.string.password),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Password,
+                        imeAction = ImeAction.Next
+                    ),
+                    keyboardActions = KeyboardActions(
+                        onNext = { localFocusManager.moveFocus(FocusDirection.Down) }
+                    ),
+                    visualTransformation = PasswordVisualTransformation(mask = '*')
+                ) { newPassword ->
+                    password = newPassword
+                }
+
+                Spacer(Modifier.height(24.dp))
+
+                AuthTextField(
```

Comment:
I think we can extract `CommonAuthField` composable and reuse it here and on Sign In. Wdyt?

Reply:
Maybe something like the last commit?🤔

Reply:
Is there a big difference between `PasswordTextField` and `ConfirmPasswordTextField`? Can't we pass that difference as a parameter? Good for `EmailTextField` 👍

Reply:
Yes, we can.

Reply:
You can see changes in the last commit. Is everything OK?